WordPress/Classes/Networking/ReaderPostServiceRemote.m

```diff
@@ -979,6 +983,52 @@ - (NSString *)formatVideoPress:(NSString *)string
     return mstr;
 }
 
+- (NSString *)resizeGalleryImageURLsForContent:(NSString *)content isPrivateSite:(BOOL)isPrivateSite
+{
+    if (!content) {
+        return @"";
+    }
+
+    NSMutableString *mcontent = [content mutableCopy];
+
+    static NSRegularExpression *regexGalleryImages;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        regexGalleryImages = [NSRegularExpression regularExpressionWithPattern:@"<img[^>]*data-orig-file[^>]*\\/>" options:NSRegularExpressionCaseInsensitive error:nil];
+    });
+
+    CGSize imageSize = [UIApplication sharedApplication].keyWindow.frame.size;
+    CGFloat scale = [[UIScreen mainScreen] scale];
+    CGSize scaledSize = CGSizeApplyAffineTransform(imageSize, CGAffineTransformMakeScale(scale, scale));
+
+    NSArray *matches = [regexGalleryImages matchesInString:mcontent options:NSRegularExpressionCaseInsensitive range:NSMakeRange(0, [mcontent length])];
+    for (NSTextCheckingResult *match in [matches reverseObjectEnumerator]) {
+        NSMutableString *imageElementString = [[mcontent substringWithRange:match.range] mutableCopy];
+        NSString *srcImageURLString = [self parseValueForAttributeNamed:@"src" inElement:imageElementString];
+        NSString *originalImageURLString = [self parseValueForAttributeNamed:@"data-orig-file" inElement:imageElementString];
+        NSURL *originalURL = [NSURL URLWithString:originalImageURLString];
+        NSURL *modifiedURL;
+
+        if (isPrivateSite) {
+            modifiedURL = [WPImageURLHelper imageURLWithSize:scaledSize forImageURL:originalURL];
+        } else {
+            modifiedURL = [PhotonImageURLHelper photonURLWithSize:imageSize forImageURL:originalURL];
+        }
+
+        // First replace the src attribute of the <img /> element
+        NSLog(@"IMG Tag before: %@", imageElementString);
+        [imageElementString replaceOccurrencesOfString:srcImageURLString
+                                            withString:modifiedURL.absoluteString
+                                               options:NSLiteralSearch
+                                                 range:NSMakeRange(0, imageElementString.length)];
+        NSLog(@"IMG Tag after: %@", imageElementString);
```

Comment:
Can we remove or replace the NSLogs?  Great for debugging but if we need to log DDLog would be better.